src/components/forms/AttributeForm.jsx

```diff
@@ -0,0 +1,182 @@
+            alignItems="center"
+            alignContent="center"
+            wrap="nowrap"
+            spacing={1}
+          >
+            {props.readOnly || props.lockId ? (
+              <Grid item>
+                <Typography variant="h6" display="inline">
+                  idField:
+                </Typography>
+                <Typography variant="h6" display="inline" color="textSecondary">
+                  &nbsp;{attributes[idAttrName]}
+                </Typography>
+              </Grid>
+            ) : (
+              <Grid item>
+                <StringField
+                  label={idAttrName}
+                  value={attributes[idAttrName]}
+                  onChange={handleSetValue}
+                  error={props.errMsg[idAttrName] ? true : false}
+                  helperText={props.errMsg[idAttrName]}
+                />
+              </Grid>
+            )}
+            {/*Key icon*/}
+            <Grid item>
+              <Tooltip title={t('modelPanels.internalId', 'Unique Identifier')}>
+                <Key
+                  fontSize="small"
+                  color="disabled"
+                  style={{ marginTop: 8 }}
+                />
+              </Tooltip>
+            </Grid>
+          </Grid>
+        </CardContent>
+
+        {Object.keys(props.dataModel).flatMap((attrName) => {
+          const attrType = props.dataModel[attrName].type;
+          const InputField = field[attrType];
+          let otherProp = {};
+          const clearButton = (
+            <InputAdornment position="end">
+              <Tooltip title="Reset">
+                <IconButton
+                  onClick={() => {
+                    handleClearButton(attrName);
+                  }}
+                  onMouseDown={(event) => event.preventDefault()}
+                >
+                  <ClearIcon />
+                </IconButton>
+              </Tooltip>
+            </InputAdornment>
+          );
+
+          if (['Integer', 'Float', 'String'].includes(attrType)) {
+            otherProp.InputProps = {
+              readOnly: props.readOnly ? true : false,
+              endAdornment: clearButton,
+            };
+          } else if (['Boolean', 'DateTime'].includes(attrType)) {
+            otherProp.disabled = props.readOnly ? true : false;
+          }
+          return props.dataModel[attrName].foreignKey ||
+            props.dataModel[attrName].isId
+            ? []
+            : [
+                <CardContent key={attrName} className={classes.cardContent}>
+                  <InputField
+                    {...otherProp}
+                    id={attrName + '-' + attrType + 'Field'}
+                    label={attrName}
+                    value={attributes[attrName]}
+                    onChange={handleSetValue}
+                    error={props.errMsg[attrName] ? true : false}
+                    helperText={props.errMsg[attrName]}
+                  />
+                </CardContent>,
```

Comment:
As with the previous suggestion, I think a `Card` element is not a good fit for rendering input elements.

Together with the other suggestions, I would recommend rendering the `InputField` components as direct children of the `form` container.

```jsx
<form className={classes.root} >
  // legend
  { props.attributes.map(({ name, type, value }) => (
      <InputField className={classes.inputField} ...otherProps />
  )}
</form>
```

If needed, `InputField` could be another abstracted component, like `Clearbutton`, which returns the appropriate input field component based on a `type` prop or similar.